src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -170,6 +186,10 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
 
             _workerRuntime = _workerRuntime ?? Utility.GetWorkerRuntime(functions);
             _functions = functions;
+
+            // feature flag and capability check will determine the value of _workerIndexing in a future worker indexing PR
+            _workerIndexing = FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagEnableWorkerIndexing);
```

Comment:
This needs be checked ahead of time here: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs#L283

Host has logic that finds functions and indexes as noted above - this is needed for status end point that lists Functions Status / Errors - That code needs to run after worker sends FUnctions Metadata. 

Also, look at https://github.com/Azure/azure-functions-host/blob/128e7feaa8fb887fc23a5d42ad7cfc5d2bf4c3fc/src/WebJobs.Script/Host/ScriptHost.cs#L336 --> This collects indexing errors per function -- any errors that are cached by the worker and sent to host need to be reflected here.